pulsar-client/src/test/java/org/apache/pulsar/client/impl/schema/JSONSchemaTest.java

```diff
@@ -368,4 +367,33 @@ public void testEncodeAndDecodeObject() throws JsonProcessingException {
         PC roundtrippedPc = jsonSchema.decode(encoded);
         assertEquals(roundtrippedPc, pc);
     }
-}
+
+    @Test
+    public void testDateAndTimestamp() {
+        JSONSchema<PC> pcJsonSchema = JSONSchema.of(PC.class);
+
+        Seller seller = new Seller("USA","oakstreet",9999);
+        PC pc = new PC("dell","g3",2020, GPU.AMD, seller);
+
+        byte[] bytes = pcJsonSchema.encode(pc);
+        Assert.assertTrue(bytes.length > 0);
+
+        Object pc2 = pcJsonSchema.decode(bytes);
+        assertEquals(pc, pc2);
+
+        GenericJsonSchema genericJsonSchema = new GenericJsonSchema(pcJsonSchema.schemaInfo);
+        GenericRecordBuilder genericRecordBuilder = genericJsonSchema.newRecordBuilder();
+        GenericRecord genericRecord = genericRecordBuilder
+                .set("brand", "dell")
+                .set("model","g3")
+                .set("year", 2020)
+                .set("gpu", GPU.AMD)
+                .set("seller", seller)
+                .build();
```

Comment:
can we add assertions that validate the list of Fields returned by genericRecord.getFields() is matching the schema ?

Reply:
Yes, for top level fields, but I don't see how to generate JsonGenericRecord for subfields, there is no way to get the schema of sub-field from the genericJsonSchema.